CruCentralCoast/Resources/ResourcesTableViewCollectionViewCell.swift

```diff
@@ -30,56 +34,64 @@ class ResourcesTableViewCollectionViewCell: UICollectionViewCell {
         self.tableView.registerCell(AudioResourcesCell.self)
         self.tableView.registerCell(VideosResourcesCell.self)
         self.tableView.registerCell(ArticlesResourcesCell.self)
+        
+        self.tableView.refreshControl = UIRefreshControl()
+        self.tableView.refreshControl?.addTarget(self, action: #selector(self.refreshResources(_:)), for: .valueChanged)
+    }
+    
+    @objc private func refreshResources(_ refreshControl: UIRefreshControl) {
+        ResourceManager.instance.refreshResources(ofType: self.type) { resources in
+            UIView.animate(withDuration: 0.3) {
+                self.resources = resources
+                self.tableView.reloadData()
+                refreshControl.endRefreshing()
+            }
+        }
     }
 }
 
 extension ResourcesTableViewCollectionViewCell: UITableViewDataSource {
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        switch self.type {
-        case .audio:
-            return 20
-        case .videos:
-            return 20
-        case .articles:
-            return 20
-        }
+        return self.resources.count
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let resource = self.resources[indexPath.row]
         switch self.type {
         case .audio:
             let cell = tableView.dequeueCell(AudioResourcesCell.self, indexPath: indexPath)
-            cell.titleLabel.text = "Audio " + String(indexPath.row)
-            cell.dateLabel.text = "Mar 17, 2018"
+            cell.titleLabel.text = resource.title
+            cell.dateLabel.text = resource.formattedDate
             return cell
-        case .videos:
+        case .video:
             let cell = tableView.dequeueCell(VideosResourcesCell.self, indexPath: indexPath)
-            cell.titleLabel.text = "Video " + String(indexPath.row)
-            cell.dateLabel.text = "Mar 17, 2018"
+            cell.titleLabel.text = resource.title
+            cell.dateLabel.text = resource.formattedDate
+            if let videoResource = resource as? VideoResource {
+                if let url = URL(string: videoResource.imageURL) {
+                    cell.previewImage.downloadedFrom(url: url)
```

Comment:
Will this show the incorrect image if the cell is reused while loading the image?

Reply:
I'm actually not sure... that's a great question... possibly yet another reason to have a completely separate image manager that deals with all of this image downloading stuff rather than having the cell or imageview itself handling the downloading.... maybe...

Reply:
True, I guess we can fix it later when we create an image manager